pkg/collector/corechecks/containerlifecycle/event.go

```diff
@@ -0,0 +1,143 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containerlifecycle
+
+import (
+	"fmt"
+
+	model "github.com/DataDog/agent-payload/v5/contlcycle"
+	types "github.com/DataDog/datadog-agent/pkg/containerlifecycle"
+)
+
+type event interface {
+	withObjectKind(string)
+	withObjectID(string)
+	withEventType(string)
+	withSource(string)
+	withContainerExitCode(*int32)
+	withContainerExitTimestamp(*int64)
```

Comment:
why `f(string)` but not `f(*string)` like you did for `(*int32)` and `(*int64)` ?
I know that exit code can be optionnel, but in that case the user should just not call `withContainerExitCode()`

Reply:
maybe the best is to move also `f(string)` to `f(*string)` (for memory allocation improvement, but maybe the compiler is already able to detect and reuse the string in the struct after)

Reply:
Yes exit codes and exit timestamps are pointers because they're optional fields. I'd keep them `f(string)` as per this [explanation](https://stackoverflow.com/questions/23542989/pointers-vs-values-in-parameters-and-return-values).
> Slices, maps, channels, strings, function values, and interface values are implemented with pointers internally, and a pointer to them is often redundant.